File: app/src/main/java/com/jinhanexample/designPattern/adapter/java/operation/DivideOperationAdapter.java
package com.jinhanexample.designPattern.adapter.java.operation;

public class DivideOperationAdapter extends AbstractOperationTarget {
    protected OperationAdaptee operationAdaptee;

    public DivideOperationAdapter(OperationAdaptee operationAdaptee) {
        this.operationAdaptee = operationAdaptee;
    }

    @Override
    public int operate(int firstNumber, int secondNumber) {
        return operationAdaptee.calculate(OperationAdaptee.DIVIDE_OPERATION, firstNumber, secondNumber);
    }
}

File: app/src/main/java/com/jinhanexample/designPattern/adapter/kotlin/operation/DivideOperationAdapter.kt
package com.jinhanexample.designPattern.adapter.kotlin.operation

class DivideOperationAdapter(private var operationAdaptee: OperationAdaptee) :
    AbstractOperationTarget() {

    override fun operate(firstNumber: Int, secondNumber: Int): Int {
        return operationAdaptee.calculate(
            OperationAdaptee.DIVIDE_OPERATION,
            firstNumber,
            secondNumber
        )
    }

}